lms/djangoapps/courseware/access.py

```diff
@@ -552,18 +553,38 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
-        response = (
-            _visible_to_nonstaff_users(descriptor)
-            and _has_group_access(descriptor, user, course_key)
-            and
-            (
-                _has_detached_class_tag(descriptor)
-                or _can_access_descriptor_with_start_date(user, descriptor, course_key)
+        has_staff_access = _has_staff_access_to_descriptor(user, descriptor, course_key)
```

Comment:
For performance reasons, return `ACCESS_GRANTED` right here if the user has staff access.  This way, the other checks below don't need to be made for staff users.